CONTRIBUTING.rst

```diff
@@ -0,0 +1,162 @@
+Contributing to the Jupyter Notebook
+====================================
+
+TODO: a welcoming sentence
+
+General Guidelines
+------------------
+
+For general documentation about contributing to Jupyter projects, see the
+`Project Jupyter Contributor Documentation`__.
+
+__ http://jupyter.readthedocs.org/#contributor-documentation
+
+
+Setting Up a Development Environment
+------------------------------------
+
+If you have already installed the dependencies mentioned below, the following
+steps should get you going::
+
+    pip install setuptools pip --upgrade --user
+    git clone https://github.com/jupyter/notebook
+    cd notebook
+    pip install -e . --user
+
+If you want this to be available for all users of your system (assuming you
+have the necessary rights) or if you are running the commands in a virtualenv,
+just drop the ``--user`` option.
+
+Installing the Dependencies
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Python Development Libraries
+""""""""""""""""""""""""""""
+
```

Comment:
Should we actually lead beginners there?

I assume that if they have Python installed already (which, if anywhere, should be documented in the README and not in CONTRIBUTING), your link doesn't help them at all, right?

I can only speak about Linux, where this link definitely doesn't help any beginner.
Is this different for OSX or Windows?